release/cloudbuild.yaml

```diff
@@ -93,31 +93,25 @@ steps:
       - gs://${_FIRMWARE_BUCKET}/${_FIRMWARE_COMPONENT}/${TAG_NAME}/trusted_applet_manifest
   ### Write the firmware release to the transparency log.
   # Copy the log entry to the sequence bucket, preparing to write to log.
+  #
+  # Use the SHA256 of the manifest as the directory to contain the entry that
+  # this trigger instance is trying to add to the log. This allows multiple
+  # triggers to run without colliding.
   - name: gcr.io/cloud-builders/gcloud
+    entrypoint: bash
     args:
-      - storage
-      - cp
-      - output/trusted_applet_manifest
-      - gs://${_LOG_NAME}/${_ENTRIES_DIR}/trusted_applet_manifest
+      - -c
+      - |
+        gcloud storage cp output/trusted_applet_manifest \
+        gs://${_LOG_NAME}/$(sha256sum output/trusted_applet_manifest | cut -f1 -d" ")/trusted_applet_manifest
```

Comment:
Is there a reason to put new top level directories all over the place? It seems like we should be able to achieve the best of both worlds by continuing to use `${_ENTRIES_DIR}` so that all pending files are together, and uniqueness by using hashing only on the filename (not the directory). e.g.:

`gs://${_LOG_NAME}/${_ENTRIES_DIR}/$(sha256sum output/trusted_applet_manifest | cut -f1 -d" ")`

If that can't work because you want each sequence event to take only a single file, then consider either making it so that the sequence command takes a single file as an alternative to a directory, OR use a subdirectory inside the ENTRIES_DIR, e.g.:

`gs://${_LOG_NAME}/${_ENTRIES_DIR}/$(sha256sum output/trusted_applet_manifest | cut -f1 -d" ")/trusted_applet_manifest`

Reply:
Nice one, changing it to your first suggestion allowed me to revert the nasty Cloud Build yaml JSON formatting.

Thinking through this change: now, each trigger will call sequence on $_ENTRIES_DIR, even though it may contain files from other triggers. This is ok because sequence will attempt to squash the dupe (https://github.com/transparency-dev/serverless-log/blob/a4ff46dcbba4213c21e46b00fd9854ae8159b273/experimental/gcp-log/internal/storage/storage.go#L265-L269). In the event that the race condition of calling Sequence results in a dupe, would that be okay for the verifier? I think it means we will have more than one manifest in the log for the same firmware.

Reply:
Cool, thanks! And great that this gets rid of the need to make other code uglier.

I trust the deduping to work. If it doesn't, having duplicates will cause a slight bit more work for everyone, but that work is trivial:
 - Believer (firmware updater): needs to download an extra manifest, compares it against the version it already has, and quits out when that version is not greater than what it currently has
 - Verifier: at the moment this will verify another manifest, and it will also check out and build. We could add logic to remember previous versions and short-circuit if seen before and both manifests are byte-equal

I'd rather not have a system where this was happening regularly though. We still have the ace up our sleeve from my last comment which is to modify the sequencer to take a single file instead of a directory (or more precisely, add a new flag which is mutually exclusive with the current sequence_dir flag). It seems like a common use-case such as this will be a file-by-file incremental approach and so this seems totally reasonable to add, and completely fixes this problem.

Reply:
Ack, I think the tweak-to-make would be here: https://github.com/transparency-dev/serverless-log/blob/588d442b0114727d87294b7f29418d38e0b0c25f/experimental/gcp-log/internal/storage/storage.go#L270-L274